### PR TITLE
feat: track availability of the transports

### PR DESCRIPTION
**Pull request by JeferssonCL**
# Pull Request

**User Story Link:** https://app.clickup.com/t/8689n1gr4

This pull request introduces a new structure for handling data persistence of the Transports table using JSON files instead of the PostgreSQL database with Dapper. New endpoints, DTOs, and data storage and update methods have been created in the JSON files, ensuring an efficient structure for handling transports in the system.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality
      to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by JorgeHB69** [comment]
Task linked: [CU-8689n1gr4 Track availability of the transports (Endpoints)](https://app.clickup.com/t/8689n1gr4)

File: src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public required int? Capacity { get; init; }
+    public required int? CurrentCapacity { get; init; }
+    public required bool? IsAvailable { get; init; }
+
+    public Transport FromEntity(Transport entity)
+    {
+        ArgumentNullException.ThrowIfNull(entity, nameof(entity));
+
+        entity.Name = Name ?? entity.Name;
+        entity.Plate = Plate ?? entity.Plate;
```
Comment (JorgeHB69) [inline src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs]: The plate should be editable?

**Review by JorgeHB69 — CHANGES_REQUESTED** [review]

**Comment by Miguelito2774** [comment]
POST

1. When adding a transport, the JSON rewrites all existing data, generating duplicates of all registered transports.

PUT

1. Currently, editing of a transport's nameplate is allowed. It would be more appropriate to limit edits to capacity and availability, leaving the nameplate and name uneditable.

File: src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs (hunk 1 above)
Comment (JeferssonCL) [reply src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs:21]: Sip porque no?

Comment (JeferssonCL) [reply src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs]: Mmm I think that no, so I will update this.

Comment (JeferssonCL) [reply src/DistributionCenter.Infraestructure/DTOs/Concretes/Transports/UpdateTransportDto.cs]: Done

File: src/DistributionCenter.Domain/DistributionCenter.Domain.csproj
Comment (MorveN11) [inline src/DistributionCenter.Domain/DistributionCenter.Domain.csproj:1]: Why is necessary add commons to the Domain?

**Review by Miguelito2774 — DISMISSED** [review]
Now yes, good job.

**Review by aguirrekarina — DISMISSED** [review]
Good job

**Review by MorveN11 — APPROVED** [review]
Good Job @JeferssonCL

**Review by JorgeHB69 — APPROVED** [review]

**Review by Miguelito2774 — APPROVED** [review]

**Pull request merged**